src/legacy/core_plugins/kibana/public/dashboard/panel/panel_utils.ts

```diff
@@ -44,10 +56,10 @@ export class PanelUtils {
       y: panel.row - 1,
       w: panel.size_x || DEFAULT_PANEL_WIDTH,
       h: panel.size_y || DEFAULT_PANEL_HEIGHT,
-      i: panel.panelIndex.toString()
+      i: panel.panelIndex.toString(),
     };
     panel.version = chrome.getKibanaVersion();
-    panel.panelIndex = panel.panelIndex.toString();
+    panel.panelIndex = panel.panelIndex;
```

Comment:
I think you're correct in dropping the `toString()` call here as it seems like `panelIndex` is already a string, but if that's the case then you should also be able to drop it in the above object (currently line 59's `i: panel.panelIndex.toString()`)

Reply:
There are some unit tests that wig out when I remove the .toString() in line 59, loads old panel data in the right order with margins and loads old panel data in the right order

Reply:
Changed the test file to input PanelIndex as a string (currently is a js file). Can also remove what was line 64 since ts is making sure that it is always going to be cast as a string so we dont need to additionally confirm it.

Reply:
Ok this was weird...It looks like the current branch failed. I was wondering if it was related to this change since it was having issues running locally before. I ran `node scripts/jest dashboard app using legacy data bwc shared urls 6.0 urls loads an unsaved dashboard` in my branch again with the .toString() included and it passed locally. When I removed the .toString() and ran the same command it failed.

Reply:
Ok looks like the commit with that change fixed the failing tests...:thinking_face_emoji:

Reply:
The failing tests are associated with changing panel_utils.ts line 59 is actually not with panel_utils.test but the dashboard_grid_container. When checking typeof in the panel_utils, it's being passed panelIndex as a string. However, in dashboard_grid_container.test.js, the test data is passing in panelIndex as numbers. In the file`get_app_State_defaults.js` you have:     `panels: savedDashboard.panelsJSON ? JSON.parse(savedDashboard.panelsJSON) : [],`
This was intentional and including the `panel.panelIndex.toString()` in panel_utils to account for this potential number type being passed in as a number. Leaving a comment in the code to get back to in a future PR.